### DISCUSSION TITLE
Error account.CreateTransaction is not a function on running deploy script

### DISCUSSION THREAD

**Discussion by wluque** [TENK-DAO/tenk#85] [General]
Hello , I am trying to run the deployment script :  

near repl -s ./scripts/deploy.ts --accountId <my account id> -- <my contract>.testnet

and I have the following error : TypeError: account.CreateTransaction is not a function

need to install something? , this script would be the first step to deploy the smart contract to near?

thank you in advance

**Comment by willemneal** [comment]
> near repl -s ./scripts/deploy.ts --accountId -- .testnet

You need to supply the account id, e.g.

> near repl -s ./scripts/deploy.ts --accountId bob.testnet

The error was that there was no account id so then `account == undefined` in the script and thus had no `CreateTransation` function.

**Reply by wluque** [reply]
thanks for your quick response!

I'm sorry I didn't write the command correctly when I pasted it here.

I first did the login in near (near login) with my account "nfts-01-ce.testnet" . login ok.

so the command executed is: 

near repl -s ./scripts/deploy.ts --accountId nfts-01-ce.testnet

Thank you!

**Reply by wluque** [reply]
I also tried the following:

near repl --accountId nfts-01-ce.testnet

> console.log(account)
Account {
  accessKeyByPublicKeyCache: {},
  connection: Connection {
    networkId: 'testnet',
    provider: JsonRpcProvider { connection: [Object] },
    signer: InMemorySigner { keyStore: [MergeKeyStore] }
  },
  accountId: 'nfts-01-ce.testnet'
}
undefined

**Reply by willemneal** [reply]
`
near repl -s ./script/deploy.ts
`
is a command for near cli that runs a script and passes the context of the execution, account of `accountId` and any args passed after `--`.

The reason for your situation is that, this repo uses a different version of `near-api-js` to use `createTransaction`.  Unfortunately this means that your global install of `near-cli` will be use a different version when importing the `Account` class, which lacks this function. This explains the error you got earlier.

> TypeError: account.CreateTransaction is not a function

Luckily `yarn` and `npm` scripts use the local installation of `near-cli` which solves this issue.

Try using the yarn script for deploying.

```
yarn deploy:testnet --accountId nfts-01-ce.testnet
```

This also has the added bonus that it will recompile your contract passing it the `testnet` feature.

**Reply by wluque** [reply]
ok , thanks , the rest of the scripts may not work then .
I will try to install the correct version of near-cli . I suspected that it could be a library version problem.
thank you very much for your help !

**Reply by willemneal** [reply]
Did you try?

```
yarn deploy:testnet --accountId nfts-01-ce.testnet
```

**Reply by wluque** [reply]
yes , i had already compiled it , the deploy also worked fine . 
I'll try to keep moving forward. 
thank you

**Comment by adamk72** [comment]
@willemneal  or @wluque, have you gotten this to work with an argument? I'm trying:
```sh
yarn deploy:testnet --accountId foo.testnet -- deploy_test.foo.testnet
```
And get an error that says:
```
Unknown argument: deploy_test.foo.testnet
```
I looked to see if maybe it was an issue with the 'near-cli' itself, but in my testing, using `near repl -s...` rather than `yarn` works fine if I take out the code near the bottom where the `tx` variable is created -- so the cli itself seems fine.

I tried the `yarn` variation with basically an empty function and it still generates the "Unknown argument" error.

I'm relatively new to NEAR, so might be missing something. While I can send this command: `yarn deploy:testnet --accountId foo.testnet` multiple times successfully, it's not like it re-initializes the contract with new variables, so this doesn't do me any good. I'd like to be able to make use of a sub-account contract through the `deploy.ts` script instead.

I've been successful with creating sub-account contracts doing this in my own automated way, but it lacks the elegance and integration of using doing this all through the "native" way.

**Reply by adamk72** [reply]
I did notice that if I:
1. Create a sub-account first;
2. Then run `yarn deploy:testnet --accountId sub_acct.foo.testnet`,

It works. However, I had to change the `owner_id` to also be the sub-account, otherwise, I get a permission error. And if I don't create the sub-account first, the error tells me the account can't be created, since it doesn't exist, meaning I have a poor understanding of what this line does:
```javascript
let contractAccount = new Account(near.connection, contractId);
```
as well as what it means to be an "owner". More reading needs to happen. 😄 📖 💻
